.github/workflows/main.yml

```diff
@@ -10,7 +10,10 @@ on:
     branches: [ develop ]
     paths-ignore:
       - 'doc/**'
-
+  # Allows you to run this workflow manually from the Actions tab
+  # this can be useful for running workflows for other branches than develop
```

Comment:
My understanding is that the main goal of `worklow_dispatch` is to allow manually triggering the workflow again. I even didn't know that we could use it to run the workflow on another branch.

See https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/.

I would rephrase this : "Allows to manually run the workflow, optionally on another branch" and maybe add the URL to the documentation.